aptos-move/framework/aptos-framework/sources/block.move

```diff
@@ -163,7 +177,16 @@ module aptos_framework::block {
     }
 
     /// Emit the event and update height and global timestamp
-    fun emit_new_block_event(vm: &signer, event_handle: &mut EventHandle<NewBlockEvent>, new_block_event: NewBlockEvent) {
+    fun emit_new_block_event(vm: &signer, event_handle: &mut EventHandle<NewBlockEvent>, new_block_event: NewBlockEvent) acquires CommitHistory {
+        if (exists<CommitHistory>(@aptos_framework)) {
+            let commit_history_ref = borrow_global_mut<CommitHistory>(@aptos_framework);
+            let idx = commit_history_ref.next_idx;
+            if (table_with_length::contains(&commit_history_ref.table, idx)) {
+                table_with_length::remove(&mut commit_history_ref.table, idx);
+            };
```

Comment:
you cannot overwrite?

Reply:
yeah I think "add" will just abort if it already exists